### PR TITLE
Add sidebar tree navigation

### PR DESCRIPTION
**Pull request by calvellido**
* Add ESlint rules, Stroustrup brace style

* Add navigation tree data JSON files
   * Available through build time and also one added to assets directory to be able to get it on runtime

* Add navigation tree data type and model
   * Add ambient declarations for NavTreeData to be able to import on compilation time
   * Also add a model to be able to work with this type of data on runtime dynamically

* Add navigation tree component
   * Add navTreeNode component, implemented recursively
   * Styled it following the material style
   * Use this new component on the sidebar replacing the old one
   * Add navigation tree data retrieval service

### CODE REVIEW

**Review by eperinan — APPROVED** [review]
LGTM

File: src/app/app.component.ts [hunk 1]
```diff
@@ -1,14 +1,29 @@
-import { Component } from '@angular/core';
+import { Component, OnInit } from '@angular/core';
+
+import { NavTreeDataService } from 'app/services/nav-tree-data.service';
+
+// import { NavTreeNode } from 'app/shared/nav-tree-node.model';
```
Comment (anamariamv) [inline src/app/app.component.ts]: Remove comment

File: src/app/app.component.ts [hunk 2]
```diff
@@ -1,14 +1,29 @@
...
+export class AppComponent implements OnInit {
 
-  links = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10];
+  public navTreeObject: NavTreeData;
 
+  // constructor(private navTreeDataService: NavTreeDataService) { }
   constructor() { }
 
+  ngOnInit() {
+    // this.navTreeDataService.getJSON()
```
Comment (anamariamv) [inline src/app/app.component.ts]: Same above

File: src/app/shared/nav-tree-data.json [hunk 3]
```diff
@@ -0,0 +1,89 @@
+{
```
Comment (anamariamv) [inline src/app/shared/nav-tree-data.json]: This file is duplicated. Why? 
I think that we should have it in a "common".

**Review by anamariamv — CHANGES_REQUESTED** [review]

File: src/app/app.component.ts (hunk 1 above)
Comment (calvellido) [reply src/app/app.component.ts]: Addressed on https://github.com/frees-io/freestyle-opscenter-webclient/pull/23/commits/3ea1e64e5dfc35d7c98dbf7f7f9f3ac9391e7607

Comment (calvellido) [reply src/app/app.component.ts]: Addressed on https://github.com/frees-io/freestyle-opscenter-webclient/pull/23/commits/3ea1e64e5dfc35d7c98dbf7f7f9f3ac9391e7607

File: src/app/app.component.ts (hunk 2 above)
Comment (calvellido) [reply src/app/app.component.ts]: Addressed on https://github.com/frees-io/freestyle-opscenter-webclient/pull/23/commits/3ea1e64e5dfc35d7c98dbf7f7f9f3ac9391e7607

File: src/app/shared/nav-tree-data.json (hunk 3 above)
Comment (calvellido) [reply src/app/shared/nav-tree-data.json]: Addressed on https://github.com/frees-io/freestyle-opscenter-webclient/pull/23/commits/3ea1e64e5dfc35d7c98dbf7f7f9f3ac9391e7607

**Review by eperinan — APPROVED** [review]
LGTM

**Review by anamariamv — APPROVED** [review]

**Pull request merged**